runtimes/protocol/lsp.ts

```diff
@@ -40,6 +41,41 @@ export class AutoParameterStructuresProtocolRequestType<P, R, PR, E, RO>
     }
 }
 
+/**
+ * Extra Client information passed during initialization handshake from Client to Server.
+ */
+export interface AWSClientInitializationOptions {
+    /**
+     * Custom client branding information.
+     */
+    product?: {
+        name: string
+        version: string
+    }
+    /**
+     * Client host platform information.
+     */
+    platform?: {
+        name: string
+        version: string
+    }
+    /**
+     * Custom Client ID value, set based on client extension internal business logic.
+     */
+    clientId?: string
+}
+
+/**
+ * Extended AWS Runtimes InitializeParams interface,
+ * sent from Client to Server as part of [`initialize`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize) request
+ */
+export interface InitializeParams extends _InitializeParamsBase {
+    initializationOptions?: {
+        [key: string]: any
+        aws: AWSClientInitializationOptions
```

Comment:
I'd like to suggest `awsClient`, as this field seems to describe the client. In the future if there was something "AWS" specific that needed to be passed along (like a directive along the lines of "prefer doing this when using service foo"), I'm under the impression this would go in a separate object. But if you intend for all of that to go into `AWSClientInitializationOptions` I have no objection. AWSClientInitializationOptions would simply become a catch-all over time.

Reply:
I'd prefer to rename this `clientInfo` instead of `aws` since that's actually what it captures? I assume we only have this field here because the LSP client in e.g., Visual Studio does not allow us to override the proper `clientInfo` in the `initializeRequest`?

Reply:
I am using `aws` as a namespace to scope all custom extension points we do to standard LSP. I'll nest proposed fields in `aws.clientInfo`, it seems to be logical grouping for these fields based on this thread.